mongo/errors.go

```diff
@@ -85,11 +85,10 @@ func replaceErrors(err error) error {
 
 // IsDuplicateKeyError returns true if err is a duplicate key error
 func IsDuplicateKeyError(err error) bool {
-	// handles SERVER-7164 and SERVER-11493
+	// handles SERVER-7164
 	for ; err != nil; err = unwrap(err) {
 		if e, ok := err.(ServerError); ok {
-			return e.HasErrorCode(11000) || e.HasErrorCode(11001) || e.HasErrorCode(12582) ||
-				e.HasErrorCodeWithMessage(16460, " E11000 ")
+			return e.HasErrorCode(11000) || e.HasErrorCode(11001) || e.HasErrorCode(12582)
```

Comment:
This internal use of `HasErrorCodeWithMessage` was a workaround for [SERVER-11493](https://jira.mongodb.org/browse/SERVER-11493). Duplicate key errors on a sharded cluster used to have an error code of 16460, a code common to a range of insert errors. So, we had to check the message as well. According to the discussion on 11493, though, sharded clusters now return a 11000 code for duplicate keys (backported to 2.4), so I think it's safe to remove this check. Note that I had to change the code in the associated test, too.

Reply:
I understand why we're making this change, but it seems out of scope for this ticket. The ticket is primarily about errors that convey SDAM state changes. Given that, I'd prefer to leave this code as-is, but let me know what you think.

Reply:
It's a fair point. I would say that the general idea of the ticket is to prefer codes over messages in drivers generally, but it looks like other drivers are only making SDAM-related changes. So, I'll leave `IsDuplicateKeyError`, `IsMaxTimeMSExpiredError` and `NamespaceNotFound` as is.